host/examples/tx_samples_from_file.cpp

```diff
@@ -221,7 +227,21 @@ int UHD_SAFE_MAIN(int argc, char* argv[])
     uhd::tx_streamer::sptr tx_stream = usrp->get_tx_stream(stream_args);
 
     // send from file
-    do {
+    while(true) {
+
+        // First check whether to send or to stop
+        if(stop_signal_called)
+            break;
+
+        if(repeat and (delay > 0.0)) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(int64_t(delay * 1000)));
```

Comment:
moving this wit here modify the behavior to wait before first execution which seems not expected to me

Reply:
Moved it back and restored the old behavior.

BTW: I made it as to not break backwards compatibility (still have the --repeat) option. It could be merged by allowing `n_repeat=-1` as substitute for `repeat`, i.e. repeating indefinitely. Which version would you prefer?